### PR TITLE
gather artifacts for e2e setup

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR adds artifact gathering for e2e setup to try and understand e2e CI falures

### CODE REVIEW

File: hack/test-e2e-olm.sh [hunk 1]
```diff
@@ -3,6 +3,28 @@
...
+test_artifactdir="${ARTIFACT_DIR}/$(basename ${BASH_SOURCE[0]})"
+if [ ! -d $test_artifactdir ] ; then
+  mkdir -p $test_artifactdir
+fi
+cleanup(){
+  local return_code="$?"
+  
+  set +e
+  if [ "$return_code" != "0" -a $get_setup_artifacts ] ; then 
+    oc get all -n openshift-operators-redhat > $testartifactdir/openshift-operators-redhat_all.txt
```
Comment (huikang) [inline hack/test-e2e-olm.sh:20]: How about adding a variable `ELASTICSEARCH_OPERATOR_NAMESPACE` like line 9 with the default value `openshift-operators-redhat`?

Comment (jcantrill) [reply hack/test-e2e-olm.sh:20]: all other tests assume it lives here.  do we have need to parameterize it?

Comment (huikang) [reply hack/test-e2e-olm.sh:20]: Sorry, I only looked at the change at line 9 and did not check the rest of the file. If so, it I agreed that there is no need to parameterize to be consistent.

**Review by alanconway — APPROVED** [review]
/lgtm

**Pull request merged**